src/spicy/runtime-support.cc

```diff
@@ -681,9 +681,11 @@ void rt::protocol_handle_close(const ProtocolHandle& handle) {
     c->analyzer->RemoveChildAnalyzer(handle.id());
 }
 
-rt::cookie::FileState* rt::cookie::FileStateStack::push() {
+rt::cookie::FileState* rt::cookie::FileStateStack::push(std::optional<std::string> fid_provided) {
     auto _ = hilti::rt::profiler::start("zeek/rt/file-stack-push");
-    auto fid = file_mgr->HashHandle(hilti::rt::fmt("%s.%d", _analyzer_id, ++_id_counter));
+    auto fid = fid_provided.value_or(file_mgr->HashHandle(hilti::rt::fmt("%s.%d", _analyzer_id, ++_id_counter)));
+    if ( find(fid) )
+        throw InvalidValue(hilti::rt::fmt("Duplicate file id %s provided", fid));
```

Comment:
We incur the duplicate check even if we asked Zeek to create a hash for us; I don't think we need to validate whether it works here ;)

```suggestion
    if (fid_provided && find(*fid_provided)) 
        throw InvalidValue(hilti::rt::fmt("Duplicate file id %s provided", *fid_provided));

    auto fid = fid_provided.value_or(file_mgr->HashHandle(hilti::rt::fmt("%s.%d", _analyzer_id, ++_id_counter)));
```

Reply:
Even with my suggestion we'd always invoke `HashHandle` (even if a value was provided). It would be great to avoid that even though it might require fatter conditionals.